### PR TITLE
JavaScript: Fix ugly formatting parens wrap binary expressions…

### LINKED ISSUE #5173: inconsistent bin with call and bin with member
**Prettier 1.14.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAKAJAZwgWzgeRgAs4AnAJTgEMAbGgT0toYBloBzANStIEsqARjTgACAPxiAOlBGyRWXAWJkmdRtTVsoXHvyGiJ0ufOx5CJChoarWHbn0HDxUmXIVnll5uu9adD-QBKVECAbiMjDFMlCxsfTTtdRwMXY3cYlSt42217PSdDV1l080zvOL885OcjN2jSrzUKxIDhQIA6MGYQ0JAAGhAIAAcYXmhMZFAeUggAdwAFHgQJlCoANwheABN+kAFSKjAAazgYAGUhw95tZBhSAFc4AevMMhh5g-YcKmQAM1pXgMAFaYAAeACEDsdTmcqHgWNc4H8AU8QJdSK9SMg9oJ6DRoLshnxYAB1bbEZAADgADAMiRBXiSDkNsUS4Ji1kiBqQ4ABHe68HkfKhfH5If40QEgV44Xi3B6ozDXdjCACK9wg8GRktRMEEZK2FKQACYBncqLwaMqAMK4b7YqDQLkge6vAAqghWEteAF8fUA)
```sh
--parser babylon
```

**Input:**
```jsx
($someOtherReallyReallyLongVariable ??
    $someOtherReallyReallyLongVariable ??
    $someOtherReallyReallyLongVariable ??
    $someOtherReallyReallyLongVariable)();
 
 ($someOtherReallyReallyLongVariable ??
    $someOtherReallyReallyLongVariable ??
    $someOtherReallyReallyLongVariable ??
    $someOtherReallyReallyLongVariable).call();
```

**Output:**
```jsx
($someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable)();

(
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable
).call();

```

**Expected behavior:**

```jsx
(
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable
)();

(
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable ??
  $someOtherReallyReallyLongVariable
).call();

```

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Fix #5173

```js
// Input
(
  aaaaaaaaaaaaaaaaaaaaaaaaa &&
  bbbbbbbbbbbbbbbbbbbbbbbbb &&
  ccccccccccccccccccccccccc &&
  ddddddddddddddddddddddddd &&
  eeeeeeeeeeeeeeeeeeeeeeeee
)();

// Prettier (stable)
(aaaaaaaaaaaaaaaaaaaaaaaaa &&
  bbbbbbbbbbbbbbbbbbbbbbbbb &&
  ccccccccccccccccccccccccc &&
  ddddddddddddddddddddddddd &&
  eeeeeeeeeeeeeeeeeeeeeeeee)();

// Prettier (master)
(
  aaaaaaaaaaaaaaaaaaaaaaaaa &&
  bbbbbbbbbbbbbbbbbbbbbbbbb &&
  ccccccccccccccccccccccccc &&
  ddddddddddddddddddddddddd &&
  eeeeeeeeeeeeeeeeeeeeeeeee
)();
```

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
@evilebottnawi 
I found the below formatting while adding tests:
[Playground Link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPAIwEMAnPAMgtwJNNwEoBuEAGhAgAcYBLaAZ2SgypCAHcACmQSCUxAG4QeAEzYhCpYmADWcGAGVOWnlADmyGKQCucdgAsYAWwA2AdTs94-I2Dj6ZnjzyngCeyOD8guwm-HCkMBKapo7EyKjEzrHsAFb8AB4AQpo6evrEjnAAMiZwaRlZILl5+iamznAAilYQ8HWZtiBGpLGk4SSEcM5qnKKxrpqc4TNwI-K17ACO3fCJXLIgxPwAtFBwcMrnaqRwWzzXicTJqUjp-eyxjjwW1gP8re1dHq1F71AYwYiEVwqGB2ZAAJnYlmIPGcrQAwhBHClwlBoOsQFZYgAVCGyV4NGYmGBQ5Qw5AARgADABfZlAA)
```js
// Input
foo(
  bar &&
  bar
);

// Output
foo(
  bar &&
    bar
);
```

Is this a bug?
(But I think if this is a bug, it should be opened as other issue.)

**Comment by evilebottnawi** [comment]
@sosukesuzuki yes i think it is bug, open new issue :+1:

**Comment by sosukesuzuki** [comment]
@evilebottnawi I've added (and fixed) tests, and created [new issue](https://github.com/prettier/prettier/issues/6442) :+1:

File: CHANGELOG.unreleased.md [hunk 1]
```diff
@@ -521,6 +553,7 @@ class Class {
 [#6412]: https://github.com/prettier/prettier/pull/6412
 [#6420]: https://github.com/prettier/prettier/pull/6420
 [#6411]: https://github.com/prettier/prettier/pull/6411
+[#]: https://github.com/prettier/prettier/pull
```
Comment (lydell) [inline CHANGELOG.unreleased.md]: Missing PR number?

Comment (sosukesuzuki) [reply CHANGELOG.unreleased.md]: Thanks, I mistook. fixed.

**Comment by suchipi** [comment]
Thanks for doing this 💜

**Pull request merged**